v3/examples/sqlite/main.go

```diff
@@ -0,0 +1,48 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	spinhttp "github.com/spinframework/spin-go-sdk/v3/http"
+	"github.com/spinframework/spin-go-sdk/v3/sqlite"
+)
+
+type Pet struct {
+	ID        int64
+	Name      string
+	Prey      *string // nullable field must be a pointer
+	IsFinicky bool
+}
+
+func init() {
+	spinhttp.Handle(func(w http.ResponseWriter, r *http.Request) {
+		db := sqlite.Open("default")
+		defer db.Close()
+
+		_, err := db.Query("REPLACE INTO pets VALUES (4, 'Maya', ?, false);", "bananas")
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		rows, err := db.Query("SELECT * FROM pets")
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		var pets []*Pet
+		for rows.Next() {
+			var pet Pet
+			if err := rows.Scan(&pet.ID, &pet.Name, &pet.Prey, &pet.IsFinicky); err != nil {
```

Comment:
I don't know Go idioms but if this is reliant on order then shouldn't the SELECT statement on line 30 force the order?